python/setup.cfg

```diff
@@ -42,6 +42,16 @@ package_dir =
     = src
 packages = find:
 python_requires = >=3.7
+install_requires =
+    pytz
+    retrying
+    mmh3
+    fastavro>=1.3.2<1.4.0
+    hmsclient==0.1.1
+    boto3
+    pyarrow
```

Comment:
A few of these will be added [here](https://github.com/apache/iceberg/pull/4262/files#diff-1f5419f88832f5bdcc37bc3ef9cfd11a714489c537227ec469756177a2b1578eR45) as part of PR #4262 and some will be part of upcoming PR's (like boto3 being added as part of a `BotoFileIO` PR and hmsclient being added as part of a `HiveCatalog` PR). I think this list can be reduced to just `pyarrow` for now.

Reply:
Good point then I think it's worth adding this to the `extras_require` block, so something like this:
```
[options.extras_require]
arrow =
    pyarrow
```
This would allow someone to do `pip3 install iceberg[arrow]` which will provide all of the iceberg core dependencies + optional arrow dependencies and as we expand this, something like `pip3 install iceberg[arrow,jdbc]`. This still allows you to just do `pip3 install iceberg` if you already have a specific pyarrow version in your environment that you want to use instead.

Reply:
Hey @samredai @rdblue Great point, and I agree. Pulling in boto might be too much if you don't use it at all, same for Hive.

Ps. Maybe later on we also might want to look into a full dependency management system for the python codebase. Something like [Poetry](https://python-poetry.org/) or [PDM](https://pdm.fming.dev/). We don't really pin versions of packages, which might cause problems when a downstream package is released with a breaking change (which unfortunately [still happens](https://github.com/dbt-labs/dbt-core/issues/4745) quite a lot).